spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +71,16 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	@PostConstruct
+	public void deduceModuleArguments() throws Exception {
```

Comment:
I put it here (and so this also justifies this class seeing PropertySources) because then this class is fully responsible for converting from Env to modules + args. IF `@ConfigurationProperties` somehow magically could bind dynamic properties from Maps of which I don't know the keys until runtime, this would "simply" be a `@ConfigurationProperties` class. Moving the `deduceModuleArguments()` logic to the launcher feels like we're leaving the `modules` bit behind. Even more so if it becomes a `Map<label, coordinates>` as expressed in another comment, don't you think?